nova_vm/src/execution/realm/intrinsics.rs

```diff
@@ -0,0 +1,237 @@
+use crate::{
+    heap::{BuiltinObjectIndexes, Handle},
+    types::{Object, Value},
+};
+
+// TODO: We should probably consider lazily loading intrinsics. This would
+//       contain a mutable reference to [`Realm`] and be created via a
+//       `Realm::intrinsic()` method to guarantee safety.
+
+pub struct Intrinsics;
+
+impl Intrinsics {
+    /// %Array%
+    pub fn array() -> Object {
+        Object::new(Value::Object(Handle::new(
+            BuiltinObjectIndexes::ArrayConstructorIndex as u32,
```

Comment:
This is incorrect: The constructor should be a function and the arrayconstructorindex is the index of the function's heap object data which the function heap data refers to. I had the `get_constructor_index` function somewhere on the heap constants file that translates the built-in object heap data index into the corresponding function index.